src/assets/AssetsDetectionController.ts

```diff
@@ -242,8 +244,14 @@ export class AssetsDetectionController extends BaseController<AssetsDetectionCon
 						true
 					);
 				}
+				collectiblesToRemove = collectiblesToRemove.filter((c) => {
+					return !(c.tokenId === Number(token_id) && c.address === toChecksumAddress(address));
+				});
 			});
 			await Promise.all(addCollectiblesPromises);
+			collectiblesToRemove.map(({ address, tokenId }) => {
```

Comment:
there's no need to return an array here. forEach would suffice.

Reply:
totally